xml/System.Runtime.CompilerServices/MethodImplOptions.xml

```diff
@@ -135,7 +135,9 @@
       </ReturnValue>
       <MemberValue>512</MemberValue>
       <Docs>
-        <summary>The method contains a hot path and should be optimized.</summary>
+        <summary>The method should skip tiered-compilation and immediately reach tier0. 
+                 This can be beneficial for certain hot-path code which does not rely on any static state or any optimisations applied at re-jit time. 
```

Comment:
Which optimizations, which paths are "certain"? As a someone who is looking to optimize my code and not sure whether I should use this attribute or not, I am afraid this is too little detail, at the same time, I do not know how to better word this...

I think the general guidance of: "this method is known to be hot, does not rely on state that is known only after is has been executed, and is on a startup path" is OK'ish?

At the same time, no Tier0 means no profile data and no PGO, so in a sense, all methods should benefit, except if they are hand-optimized with `goto`s and such.